File: javascript_client/src/storage.js
export class Storage {
  constructor(client) {
    this._client = client;
  }

  tables() {
    return this._client._fetch('/v1/tables/storage', { auth: 'auto' });
  }

  objects() {
    return this._client._fetch('/v1/buckets/storage', { auth: 'auto' });
  }
}

File: kotlin_client/src/main/kotlin/org/ezauth/client/Storage.kt
package org.ezauth.client

/** Storage usage. */
class Storage internal constructor(private val client: BaseClient) {

    suspend fun tables(): StorageResponse {
        val resp = client.fetch("/v1/tables/storage", auth = AuthMode.AUTO)
        return client.json.decodeFromJsonElement(StorageResponse.serializer(), resp!!)
    }

    suspend fun objects(): StorageResponse {
        val resp = client.fetch("/v1/buckets/storage", auth = AuthMode.AUTO)
        return client.json.decodeFromJsonElement(StorageResponse.serializer(), resp!!)
    }
}
